console/src/views/Radio.js

```diff
@@ -0,0 +1,136 @@
+import ConsoleToggleButton from '../components/ConsoleToggleButton';
+
+import css from './Radio.less';
+
+
+const RadioBase = kind({
+  name: 'Radio',
+
+  handlers: {
+    onFrequencyToggle: (ev, {changeFrequency}) => {
+      const frequency = ev.currentTarget.innerText.trim();
+      changeFrequency(frequency);
+    },
+    onPresetClick: (ev, {changeStation, presets}) => {
+      const index = parseInt(ev.currentTarget.getAttribute('data-preset'));
+      changeStation(presets[index]);
+    },
+    onTune: (ev, {currentStation, changeStation}) => {
+      const action = ev.currentTarget.getAttribute("data-action");
+      let newStation;
+
+      if (action === 'tune-up') {
+        if (currentStation >= 108) {
+          newStation = 87.8;
+        } else {
+          newStation = (currentStation*10 + 1)/10;
+        }
+      } else if (action === 'tune-down') {
+        if (currentStation <= 87.8) {
+          newStation = 108;
+        } else {
+          newStation = (currentStation*10 - 1)/10;
+        }
+      } else if (action === 'scan-up') {
+        if (currentStation >= 108) {
+          newStation = 87.8;
+        } else {
+          newStation = (currentStation*10 + 20)/10;
+        }
+      } else if (action === 'scan-down') {
+        if (currentStation <= 87.8) {
+          newStation = 108;
+        } else {
+          newStation = (currentStation*10 - 20)/10
+        }
+      }
+      changeStation(newStation);
+    }
+  },
+
+  render: ({currentStation, frequency, presets, onFrequencyToggle, onPresetClick, onTune}) => {
+
+    return (
+      <Panel>
+        <div className={css.radio}>
+          <div className={css.radioToggle}>
+            <ConsoleToggleButton onClick={onFrequencyToggle} selected={frequency === 'AM'} small type="grid">AM</ConsoleToggleButton> |
+            <ConsoleToggleButton onClick={onFrequencyToggle} selected={frequency === 'FM'} small type="grid">FM</ConsoleToggleButton>
+          </div>
+          <div className={css.title}>
+            {/*Radio TextInfo*/}
+            <LabeledItemBase label="Artist - Song">{currentStation} MHZ</LabeledItemBase>
+          </div>
+          <div className={css.tuneControls}>
+            {/*Tune*/}
+            <ConsoleButton onClick={onTune} data-action="tune-down" icon={"arrowsmallleft"} />
+            Tune
+            <ConsoleButton onClick={onTune} data-action="tune-up" icon={"arrowsmallright"} />
+          </div>
+          <div className={css.scanControls}>
+            <ConsoleButton onClick={onTune} data-action="scan-down" icon={"arrowsmallleft"} />
+            Scan
+            <ConsoleButton onClick={onTune} data-action="scan-up" icon={"arrowsmallright"} />
+          </div>
+          {/*List*/}
+          <div className={css.presetList}>
+            Presets
+            <Divider startSection />
+            <PresetItem data-preset="0" label="Station 1" icon="plus" onClick={onPresetClick}>{presets[0]} MHZ</PresetItem>
```

Comment:
We're missing an opportunity here by using a custom component but then relying on `data-` attributes to set contextual data. `PresetItem` should include a prop for the data and pass it back via a custom event.